src/collective/easyform/browser/view.py

```diff
@@ -346,6 +346,22 @@ def __call__(self, fname=None, fset=None):
         return super(EasyFormInlineValidationView, self).__call__(fname, fset)
 
 
+class GetSaveDataAdaptersView(BrowserView):
+
+    def __call__(self, *args, **kwargs):
+        """return all contained save data adapters"""
+        from collective.easyform.api import get_actions
+        from collective.easyform.interfaces import ISaveData
+        self.context = EasyFormForm(self.context, self.request)
+        form = self.context.context
+        adapters = []
+        for action_id in get_actions(form):
+            action = get_actions(form)[action_id]
```

Comment:
Why is `get_actions` called two times? Wouldn't  something like 

```
all_actions = get_actions(form)
adapters = [action[action_id] for action_id in all_actions if ISaveData.providedBy(action[action_id])]
```
do the trick?

Reply:
It did - thx for noticing